### PR TITLE
Add UML diagram for HelpCommand

### PR DESCRIPTION
**Pull request by yixiann**

### CODE REVIEW

File: docs/_dg/Implementation.md [hunk 1]
```diff
@@ -44,3 +44,19 @@ Currently, sorting is performed ascendingly. A future implementation will allow
...
+#### How the `help` command works
+1. The user specifies a specific command that they need help with. This is done using the word needed to execute a particular command in FoodRem.
+2. If this is not provided, a general help message will be shown.
+3. The provided command is cross-referenced with all available commands in FoodRem. If the command that the user needs help with does not exist, an error would be thrown. This informs the user that the command does not exist. A general help is also shown to the user.
+4. The help is shown in a new window that will open upon successful execution of the command.
+
+#### Why it is implemented this way
+When the `HelpCommand` is executed we want users to receive help immediately instead of searching for it the user guide.
+
+![HelpSequenceDiagram](../images/HelpSequenceDiagram.png)
```
Comment (Eugenetanwl3881) [inline docs/_dg/Implementation.md:60]: For the image address, @RichDom2185  mentioned that for the images, don't need to use relative file addressing (using ..). Using the direct address (images/filename) will do

**Review by Eugenetanwl3881 — APPROVED** [review]
Just a comment on the file address. Once fixed, the rest LGTM!

File: docs/_dg/Implementation.md (hunk 1 above)
Comment (yixiann) [reply docs/_dg/Implementation.md]: Thanks, didnt realise

**Review by Eugenetanwl3881 — DISMISSED** [review]
LGTM

File: docs/_dg/Implementation.md [hunk 2]
```diff
@@ -46,3 +46,19 @@ Currently, sorting is performed ascendingly. A future implementation will allow
...
 Notably, the `SortedList` wraps around a `FilteredList`, which wraps around an immutable `ObservableList`.
+
+#### Receiving help for a command
+The `help` feature provides a user with instructions of how to use certain commands.
+
+#### How the `help` command works
+1. The user specifies a specific command that they need help with. This is done using the word needed to execute a particular command in FoodRem.
+2. If this is not provided, a general help message will be shown.
+3. The provided command is cross-referenced with all available commands in FoodRem. If the command that the user needs help with does not exist, an error would be thrown. This informs the user that the command does not exist. A general help is also shown to the user.
+4. The help is shown in a new window that will open upon successful execution of the command.
```
Comment (RichDom2185) [inline docs/_dg/Implementation.md:57]: Let's follow the module's markdown style guide and use generic list item numbers.

```suggestion
1. The user specifies a specific command that they need help with. This is done using the word needed to execute a particular command in FoodRem.
1. If this is not provided, a general help message will be shown.
1. The provided command is cross-referenced with all available commands in FoodRem. If the command that the user needs help with does not exist, an error would be thrown. This informs the user that the command does not exist. A general help is also shown to the user.
1. The help is shown in a new window that will open upon successful execution of the command.
```

**Review by RichDom2185 — APPROVED** [review]
LGTM with minor style issues that can be auto-fixed by the linter, so no worries!

**Review by RichDom2185 — APPROVED** [review]
LGTM!

**Pull request merged**